owsi-core/owsi-core-components/owsi-core-component-jpa/src/main/java/fr/openwide/core/jpa/batch/executor/MultithreadedBatch.java

```diff
@@ -0,0 +1,132 @@
+package fr.openwide.core.jpa.batch.executor;
+
+import java.math.BigDecimal;
+import java.util.Date;
+import java.util.List;
+import java.util.concurrent.Callable;
+import java.util.concurrent.TimeUnit;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.config.ConfigurableBeanFactory;
+import org.springframework.context.annotation.Scope;
+import org.springframework.stereotype.Component;
+import org.springframework.transaction.TransactionStatus;
+import org.springframework.transaction.support.TransactionCallback;
+
+import com.google.common.collect.Lists;
+
+import fr.openwide.core.jpa.batch.processor.ThreadedProcessor;
+import fr.openwide.core.jpa.batch.util.ProcessorProgressLogger;
+import fr.openwide.core.spring.property.service.IPropertyService;
+
+@Component
+@Scope(ConfigurableBeanFactory.SCOPE_PROTOTYPE)
+public class MultithreadedBatch extends AbstractBatch<MultithreadedBatch> {
+	
+	private static final Logger LOGGER = LoggerFactory.getLogger(MultithreadedBatch.class);
+	
+	private static final Logger PROGRESS_LOGGER = LoggerFactory.getLogger(ProcessorProgressLogger.class);
+	
+	@Autowired
+	private IPropertyService propertyService;
+	
+	private int threads = 2;
+	
+	private int timeoutInMinutes = 15;
+	
+	public MultithreadedBatch threads(int threads) {
+		this.threads = threads;
+		return this;
+	}
+	
+	public MultithreadedBatch timeoutInMinutes(int timeout) {
+		this.timeoutInMinutes = timeout;
+		return this;
+	}
+	
+	public void run(String context, final List<Long> entityIds, final BatchRunnable<Long> batchRunnable) {
+		Date startTime = new Date();
+		
+		LOGGER.info("Beginning batch for %1$s: %2$d objects", context, entityIds.size());
+		
+		LOGGER.info("    preExecute start");
+		
+		writeTransactionTemplate.execute(new TransactionCallback<Void>() {
```

Comment:
I wonder if it's a good idea to make these processings read/write by default... But on the other hand, making it read-only by default will surely cause frustration for developpers (when they'll find out that their runnable does not do anything, and spend an hour figuring out why).

Couldn't we provide two methods, `runReadOnly()` and `runReadWrite()`? Or something similar, but with prettier names?